### ISSUE TITLE
[quotes] False negative for backticks without neither substitution nor non-escaped quote characters  

### ISSUE THREAD

**Issue by devuxer** [typescript-eslint/typescript-eslint#4514]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
        quotes: "off",
        "@typescript-eslint/quotes": ["error", "double", { avoidEscape: true, allowTemplateLiterals: true }],
  }
}
```

```TS
    const foo = `foo`;
```

**Expected Result**

Error saying backticks not allowed. Auto-fix should replace with double quotes.

**Actual Result**

No error.

**Additional Info**

Backticks should only be allowed in these cases:

1. Expression with template tag:
    ```ts
    sql`hello`
    ```
2. Contains substitution like `${sub}`
3. Contains a quote character that would otherwise need escaping (" or ')

None of these apply in the repro.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.33.0` |
| `@typescript-eslint/parser`        | `4.33.0` |
| `TypeScript`                       | `4.5.2` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `14.16.0` |

**Comment by bradzacher** [comment]
From the docs (https://eslint.org/docs/rules/quotes#options)

> "allowTemplateLiterals": true allows strings to use backticks

This is working as expected.

**Comment by devuxer** [comment]
@bradzacher,

Thanks for your reply, but I don't agree.

The docs also say:

> "double" (default) requires the use of double quotes wherever possible

How do you get the above to be respected *except* in the case where you have `${sub}` in your string?

Without this, the same codebase can have a quoted string and backtick'd string. The whole point of a linter is to enforce consistency of code style. The way it works now, I see now way to achieve this goal.
